### PR TITLE
Add targets to Makefile to give more flexibility for the pipeline

### PR DESCRIPTION
**Pull request by oterral**
WARNING: With this PR deploying old snapshots will be impossible

Impact make targets: 
 - Add targets `build`, `dev`, `int`, `prod` : build the pap without linting and testing.
 - Add target `make flushvarnish DEPLOY_TARGET=<dev|int|prod|infra>` 
 - ` make s3deploybranch`  has been replaced by `make s3deploybranchint`

Now the pipeline takes 4 mins instead of 14

<jenkins>[Test link](https://mf-geoadmin3.int.bgdi.ch/teo_build2/index.html)</jenkins>

### CODE REVIEW

**Comment by loicgasser** [comment]
Jenkins tests are failing
```
.build-artefacts/python-venv/bin/python test/saucelabs/test.py --url https://mf-geoadmin3.int.bgdi.ch/teo_build2/index.html \
                                     --tests false \
                                     --browser chrome \
                                     --single false
Traceback (most recent call last):
  File "test/saucelabs/test.py", line 14, in <module>
    from tooltip_test import runTooltipTest
ImportError: cannot import name runTooltipTest
Makefile:269: recipe for target 'saucelabs' failed
```
Otherwise I appriciate the simplification and this should speed up the build.

File: test/saucelabs/test.py [hunk 1]
```diff
@@ -104,35 +122,27 @@ def parse_args(args):
             print 'Please try again...'
             sys.exit(1)
 
-    # if 3rd parameter equal 'true' use only top browser
-    caps_used = [top_browser] if singlebrowser else desired_cap_list
-
-    for current_desired_cap in caps_used:
+    for browser in browsers:
```
Comment (loicgasser) [inline test/saucelabs/test.py]: not ideal to use the same namespace as one of your `argument`

File: test/saucelabs/test.py [hunk 2]
```diff
@@ -71,10 +81,18 @@ def parse_args(args):
...
     ]
 
+    if single:  # The user wants to test the most used browser
+        browsers = [browsers[0]]
+    elif browser:  # The user specified the browser to test
+        bs = []
+        for b in browsers:
+            if b['browserName'] == browser:
+                bs.append(b)
+        browsers = bs
```
Comment (loicgasser) [inline test/saucelabs/test.py]: Ok alternatively you could do:
```python
browsers = [b for b in browsers if b['browserName'] == browser]
```

File: Makefile [hunk 3]
```diff
@@ -175,15 +189,32 @@ help:
 	@echo "- DEPLOY_GIT_BRANCH           (current value: $(DEPLOY_GIT_BRANCH))"
 	@echo "- GIT_COMMIT_HASH             (current value: $(GIT_COMMIT_HASH))"
 	@echo "- VERSION                     (build with: $(LAST_VERSION), current value: $(VERSION))"
+	@echo "- VARNISH_HOSTS               (build with: $(LAST_VERSION), current value: ${ARNISHHOSTS})"
```
Comment (loicgasser) [inline Makefile]: typo: ```${ARNISHHOSTS}```

File: Makefile [hunk 4]
```diff
@@ -337,6 +373,19 @@ s3deleteint: guard-S3_VERSION_PATH .build-artefacts/requirements.timestamp
...
+flushvarnish: guard-DEPLOY_TARGET
+	source rc_${DEPLOY_TARGET} && make flushvarnishinternal
+
+# This internal target has been created to have the good global variable values
+# from rc_XXX file.
+flushvarnishinternal: guard-VARNISH_HOSTS guard-API_URL guard-E2E_TARGETURL
+	for VARNISHHOST in $(VARNISH_HOSTS) ; do \
+		./scripts/flushvarnish.sh $$VARNISHHOST "$(subst //,,$(API_URL))" ;\
+		./scripts/flushvarnish.sh $$VARNISHHOST "$(subst https://,,$(E2E_TARGETURL))" ;\
+		echo "Flushed varnish at: $$VARNISHHOST" ;\
```
Comment (loicgasser) [inline Makefile]: use `@echo` instead

File: test/saucelabs/test.py (hunk 2 above)
Comment (oterral) [reply test/saucelabs/test.py]: Nicer

File: Makefile (hunk 4 above)
Comment (oterral) [reply Makefile]: Doesn't work with @echo

File: Makefile (hunk 3 above)
Comment (oterral) [reply Makefile]: fixed

File: test/saucelabs/test.py (hunk 1 above)
Comment (oterral) [reply test/saucelabs/test.py]: fixed

**Pull request merged**